src/gui/editors/AutomationEditor.cpp

```diff
@@ -492,162 +419,222 @@ void AutomationEditor::drawLine( int x0In, float y0, int x1In, float y1 )
+							MidiTime newTime = m_pattern->setDragValue(valuePos,
+								level, true, mouseEvent->modifiers() & Qt::ControlModifier);
+
+							// Set the iterator to our newly created node so we can use it later
+							// for the m_moveXOffset calculation
+							clickedNode = tm.find(newTime);
+						}
+						else // If we clicked over an existing node
+						{
+							// Simply start moving/draging it
+							MidiTime newTime = m_pattern->setDragValue(MidiTime(POS(clickedNode)),
+								level, true, mouseEvent->modifiers() & Qt::ControlModifier);
+
+							// We need to update our iterator because setDragValue removes the node that
+							// is being dragged, so if we don't update it we have a bogus iterator
+							clickedNode = tm.find(newTime);
+						}
+
+						// Set the action to MOVE_VALUE so moveMouseEvent() knows we are moving a node
+						m_action = MOVE_VALUE;
+
+						// Calculate the offset from the place the mouse click happened in comparison
+						// to the center of the node
+						int alignedX = (int) ((float) ((POS(clickedNode) - m_currentPosition) * m_ppb) /
+									MidiTime::ticksPerBar());
+						m_moveXOffset = x - alignedX - 1;
+
+						// Set move-cursor
+						QCursor c(Qt::SizeAllCursor);
+						QApplication::setOverrideCursor(c);
+
+						// Update the last clicked position so it can be used if we draw a line later
+						m_drawLastTick = posTicks;
+						m_drawLastLevel = level;
+					}
+
+					Engine::getSong()->setModified();
 				}
+				else if (m_mouseDownRight) // Right click on DRAW mode erases values
+				{
+					// Update the last clicked position so we remove all nodes from
+					// that point up to the point we release the mouse button
+					m_drawLastTick = posTicks;
+
+					// If we right-clicked a node, remove it
+					if (clickedNode != tm.end())
+					{
+						m_pattern->removeValue(POS(clickedNode));
+						Engine::getSong()->setModified();
+					}
 
-				// move it
-				m_action = MOVE_VALUE;
-				int aligned_x = (int)( (float)( (
-						it.key() -
-						m_currentPosition ) *
-						m_ppb ) / MidiTime::ticksPerBar() );
-				m_moveXOffset = x - aligned_x - 1;
-				// set move-cursor
-				QCursor c( Qt::SizeAllCursor );
-				QApplication::setOverrideCursor( c );
-
-				Engine::getSong()->setModified();
+					m_action = ERASE_VALUES;
+				}
+			break;
 			}
-			else if( ( mouseEvent->button() == Qt::RightButton &&
-							m_editMode == DRAW ) ||
-					m_editMode == ERASE )
+			case ERASE:
 			{
-				m_drawLastTick = pos_ticks;
 				m_pattern->addJournalCheckPoint();
-				// erase single value
-				if( it != time_map.end() )
+
+				// On erase mode, left click removes nodes
+				if (m_mouseDownLeft)
 				{
-					m_pattern->removeValue( it.key() );
-					Engine::getSong()->setModified();
+					// Update the last clicked position so we remove all nodes from
+					// that point up to the point we release the mouse button
+					m_drawLastTick = posTicks;
```

Comment:
I decided for not moving it outside the block for now, because of the behavior mentioned (middle click updating the `m_drawLastTick` even though it doesn't do anything).